src/util/customHooks/overlayfunc.tsx

```diff
@@ -2,33 +2,30 @@
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 
 import { Dispatch, SetStateAction, useState } from "react";
+import { ActionData, ExperienceData } from "../types";
 
 /*
  *custom hook for overlay, action list
  */
 
-const useActionList = (experienceData: {
-  initializationData: { actionList: Array<[string, string, string]> };
-}) => {
+const useActionList = (experienceData: ExperienceData) => {
   const [actionList, setActionList] = useState(
-    experienceData !== undefined
-      ? experienceData.initializationData.actionList
-      : []
+    experienceData !== undefined ? experienceData.experience.actionList : []
   );
 
   const addActionToList = (
-    [path, input, assetID]: [string, string, string],
-    hookinput: Array<[string, string, string]>,
-    functioninput: Dispatch<SetStateAction<Array<[string, string, string]>>>
+    actionData: ActionData,
+    hookinput: ActionData[],
+    functioninput: Dispatch<SetStateAction<ActionData[]>>
   ) => {
-    hookinput.push([path, input, assetID]);
+    hookinput.push(actionData);
     functioninput([...actionList]);
   };
 
   const removeActionFromList = (
     index: number,
-    hookinput: Array<[string, string, string]>,
-    functioninput: Dispatch<SetStateAction<Array<[string, string, string]>>>
+    hookinput: ActionData[],
+    functioninput: Dispatch<SetStateAction<ActionData[]>>
   ) => {
     hookinput.splice(index, 1);
     functioninput([...hookinput]);
```

Comment:
Maybe we can put this entire file into the custom hook folder since you put the previewoverlayfunc.tsx in that file. To keep everything consistent. 
Also does the test import have changed for these file location changed?
Else wise lgtm.